KakapoExample/KakapoTests/SerializerTests.swift

```diff
@@ -0,0 +1,155 @@
+        let user = User(name: "Alex")
+        
+        describe("Serialization of Serializable entities") {
+            it("produce a dictionary where properties are keys and values are values") {
+                let serialized = serialize(user)
+                expect(serialized["name"] as? String).to(equal("Alex"))
+            }
+            
+            it("serialize arrays") {
+                let friend = Friend(friends: [user])
+                let serialized = serialize(friend)
+                let friends = serialized["friends"] as? [Any]
+                let first = friends?.first as? [String: Any]
+                expect(first?.keys.first).to(equal("name"))
+                expect(first?.values.first as? String).to(equal("Alex"))
+                expect(friends?.count).to(be(1))
+            }
+        }
+        
+        describe("recursive serialization") {
+            func checkObject(objects: [Any]?) {
+                let first = objects?.first as? [String: Any]
+                expect(first?.keys.first).to(equal("name"))
+                expect(first?.values.first as? String).to(equal("Alex"))
+                expect(objects?.count).to(be(1))
+            }
+            
+            it("serialize arrays and entities inside it") {
+                let friend = Friend(friends: [user])
+                let serialized = serialize(friend)
+                let friends = serialized["friends"] as? [Any]
+                checkObject(friends)
+            }
+            
+            it("serialize dictionary and entities inside it") {
+                let friend = Friend(friends: [user])
+                let serialized = serialize(friend)
+                let dictionary = serialized["dictionary"] as? [String: Any]
+                let friends = dictionary?["test"] as? [Any]
+                checkObject(friends)
+            }
+        }
+        
+        describe("Property policy serialization") { 
+            it("is not serialized if nil") {
+                let empty = MaybeEmpty(IgnorableNilProperty<Int>(nil))
+                let serialized = serialize(empty)
+                expect(serialized.count).to(be(0))
+            }
+            
+            it("is serialized if not nil") {
+                let notEmpty = MaybeEmpty(IgnorableNilProperty(1))
+                let serialized = serialize(notEmpty)
+                let value = serialized["value"] as? Int
+                expect(value).to(be(1))
+            }
+            
+            it("recursively serialize the object if needed") {
+                let notEmpty = MaybeEmpty(IgnorableNilProperty(user))
+                let serialized = serialize(notEmpty)
+                let value = serialized["value"] as? [String: Any]
+                expect(value?["name"] as? String).to(equal("Alex"))
+            }
+
+            it("recursively serialize IgnorableNilProperties") {
+                let notEmpty = MaybeEmpty(IgnorableNilProperty(IgnorableNilProperty(1)))
```

Comment:
Doesn't make sense for IgnorableNil but might make sense for propertyPolicy with different logic